### DISCUSSION TITLE
Help. Trying to retrieve the data from draft.js to react hook form

### DISCUSSION THREAD

**Discussion by esuarezz** [react-hook-form/react-hook-form#2355] [Q&A]
Hi guys!

So I have a draft.js component to introduce into react hook form and I cant get the value back to fetch it into the form, 
it is something like this:
```
  <Controller
      as={
        <>
      <..........>
      <Editor
            editorState={editorState}
            onChange={xxxx}
       />
     </>
}
  control={control}
  name={name}
/>
```
when I am writing I am getting the onChange working so I am updating the editorState updated
to get the value in this case is something like this:
editorState.getCurrentContent().getPlainText()

So I am trying to retrieve using a controller but I cant get back to be fetched into the form.

I saw something similar for a calendar where it worked using valueName, but I cant make it work in this example(Also not sure if this continue being right in the latest release)

**Comment by bluebill1049** [comment]
have you tried with Controller render v6? probably a lot easier.

**Reply by esuarezz** [reply]
I will try and will let you know! thanks!

**Comment by esuarezz** [comment]
Hi @bluebill1049 ,

shall we add some examples managing with the onchange or the value prop?
because to be honest I sorted it sending as param serValue, because It is not clear how to do it with these props, can u provide me an example? I am trying to manage it with draft.js and this is the editor where "editorState" is the value to getBack:

    <Editor
            editorState={editorState}
            blockStyleFn={myBlockStyleFn}
            handleKeyCommand={handleKeyCommand}
            onChange={handleOnChange}
          />

**Reply by bluebill1049** [reply]
hi, have you tried `render` prop if you are using v6 with Controller?

**Comment by esuarezz** [comment]


**Reply by bluebill1049** [reply]
The doc: https://react-hook-form-website-3g3ria9qa.vercel.app/api#Controller

and a codesandbox in the doc with lots of examples (render): https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Comment by esuarezz** [comment]
But I dont remember to see these examples using the render props

**Reply by bluebill1049** [reply]
they r now, i will do a drafjs in the example soon, but please read the doc there are a lot of good info there.

**Comment by esuarezz** [comment]
Hi @bluebill1049 , I saw you added the example, It is not working  in our context we are working with storybook and next(But now we are trying it in storybook;  I get "editorState.getCurrentContent is not a function" , something I can do for that?
I copied and pasted your example and also I tried to do some changes but got that,

So I had to do it setting with setValues, but I would like to do it like you said

**Reply by esuarezz** [reply]
could you take a look into this? thanks!

**Comment by esuarezz** [comment]
Hi @bluebill1049 ! did you experimented the issue I had? Thanks! I just copied and pasted your solution and I got the error I told you

**Reply by bluebill1049** [reply]
no, I didn't get the issue, it's working as you can see in the CSB. where did you implement the solution? is there a codesadnbox?

**Reply by esuarezz** [reply]
Hu @bluebill1049 !



Thank you so much to help us! we are in love with react-hook-forms, good job!

I will send you the sandbox, since I already did it but its the implementation with setValues, I will send you in some hours, thanks!

**Reply by esuarezz** [reply]
Hi! I bring here all the information I could collect!

[Example in sandBox with the error ](https://codesandbox.io/s/nextjs-5x8ll?file=/src/Draft.js)

Also I found this in draft-js for universal solution:
[https://github.com/facebook/draft-js/tree/master/examples/draft-0-10-0/universal](https://github.com/facebook/draft-js/tree/master/examples/draft-0-10-0/universal)

In the sandbox we can find next.js, latest version of react hook implementation with Controller as the example you have.

It seems the error is coming when we use the "value" from contoller props , in draftjs universal we have to create the "empy state with this":

`//Universal draftJs solution
/*
const emptyContentState = convertFromRaw({
  entityMap: {},
  blocks: [
    {
      text: "",
      key: "foo",
      type: "unstyled",
      entityRanges: []
    }
  ]
});
*/`

When normally without using react hook form we do:
`  /*const [editorState, setEditorState] = useState(
    EditorState.createEmptyState()
  );*/`

But when we are using this controller, so we dont do the "createEmptyState"(where standard solution, we dont do createContentState using the convertFromRaw, we use directly your "value" prop, and that is giving the error I guess.

I have done my best to bring the right context and the research I did, let me know your thoughts!

Thanks @bluebill1049 !

**Reply by bluebill1049** [reply]
I think you are missing `defaultValue` at the Controller  `EditorState.createEmptyState()`
